pyseq.py

```diff
@@ -879,3 +908,47 @@ def get_sequences(source):
     log.debug('time: %s' % (datetime.now() - start))
 
     return list(seqs)
+
+
+def walk(source, level=-1, topdown=True, onerror=None, followlinks=False):
+    """ Generator that traverses a directory structure starting at
+        source looking for sequences.
+        :param source: valid folder path to traverse
+        :param level: int, if < 0 traverse entire structure otherwise
+                      traverse to given depth
+        :param topdown: walk from the top down
+        :param onerror: callable to handle os.listdir errors
+        :param followlinks: whether to follow links
+    """
+    start = datetime.now()
+    assert isinstance(source, basestring) is True
+    assert os.path.exists(source) is True
+    source = os.path.abspath(source)
+    for root, dirs, files in os.walk(source, topdown, onerror, followlinks):
+        if topdown is True:
+            parts = root.replace(source, "").split(os.sep)
+            while "" in parts:
+                parts.remove("")
+            if len(parts) == level - 1:
+                del dirs[:]
+
+        files.sort(key=_natural_key)
+        seqs = []
```

Comment:
I'd like to see what caused that bug.  The reason I didn't use get_sequences is the same reason I used the natural_sort.  I think it's faster to pre sort the file names and instead searching all found seqs for a file's membership.  But that would be handled differently with your groups work.